internal/pkg/handlers/deployment.go

```diff
@@ -0,0 +1,78 @@
+package handlers
```

Comment:
nit: To keep stuff tidy and as they've quite coupled I think it would be best to sit this in the existing `deployments.go` file. We've done this in other areas of the codebase and it's quite commmon in go to group related structs into files. What do you think?

Reply:
Done. (I actually started that way and then changed it!)